### ISSUE TITLE
Refactor to use xarray instead of pandas

### ISSUE THREAD

**Issue by rsignell-usgs** [pyoceans/pocean-core#55]
How much work would it take to refactor `pocean-core` to use `xarray` instead of `pandas` as the core data model? 

Pandas seemed reasonable when this project started, but now Xarray seems a much better fit for NetCDF DSG than Pandas, since it captures the NetCDF data model, handling multidimensional data with global and variable attributes. 

How about we do this at the IOOS Code Sprint Oct 8-10?  😸

**Comment by kwilcox** [comment]
**tl;dr** don't fix what isn't broken

I'd say refactoring for the sake of refactoring isn't a big enough sell for me. There is a 2-year old [xarray branch](https://github.com/pyoceans/pocean-core/commit/0d39d1c67950c707f45c56af7c1e2924c6d99c1d) that I played with to see if `xarray` could easily "understand" the DSG types from pandas data frames. I don't recall why I stopped working on the branch, but I remember thinking I was going to end up writing mode code that was less understandable to outsiders to get the same desired outcome we already had. [This notebook](https://github.com/pyoceans/pocean-core/blob/0d39d1c67950c707f45c56af7c1e2924c6d99c1d/Untitled.ipynb) shows how far I got.

`pocean` does 3 things really well:

1.  Read a flat DataFrame (they key here is flat, as in loaded from a CSV file, no indexes) and allow a user to assign columns to axes and then export any`*` DSG type they want.
2.  Export any`*` DSG netCDF file to a flat DataFrame.
3.  Full-circle no-data-or-metadata-loss DSG -> DataFrame -> DSG and DataFrame -> DSG -> DataFrame. I don't remember why this was a design goal, but lots of decisions were made to make sure that this was possible.

I'd love it if `xarray` was the basis for 2., but it works as is... PRs accepted!

`*` - not any, but most

**Comment by rsignell-usgs** [comment]
@kwilcox, okay, this makes sense to me.   We can leave this issue open.
